### PR TITLE
Switch to Zipkin v2 span format

### PR DESCRIPTION
**Pull request by ykitamura-mdsol**
Use the simplified Zipkin v2 span format to send traces to Zipkin server.

See https://github.com/openzipkin/zipkin/issues/1499 for details

@cabbott @jcarres-mdsol @jfeltesse-mdsol @ssteeg-mdsol @piao-mdsol @adriancole

### CODE REVIEW

File: lib/zipkin-tracer/hostname_resolver.rb [hunk 1]
```diff
@@ -28,24 +28,19 @@ def resolved_ip_address?(ip_string)
         ip_string.to_i.to_s == ip_string
     end
 
-    def each_annotation(spans, &block)
+    def each_endpoint(spans, &block)
```
Comment (adriancole) [inline lib/zipkin-tracer/hostname_resolver.rb]: maybe add a comment about what this is doing? I'm guessing it is lazy parsing?

File: lib/zipkin-tracer/rack/zipkin-tracer.rb [hunk 2]
```diff
@@ -55,12 +55,11 @@ def annotate_plugin(span, env, status, response_headers, response_body)
 
     def trace!(span, zipkin_env, &block)
       trace_request_information(span, zipkin_env)
-      span.record(Trace::Annotation::SERVER_RECV)
+      span.kind = Trace::Span::Kind::SERVER
```
Comment (adriancole) [inline lib/zipkin-tracer/rack/zipkin-tracer.rb]: is there a way to tell if the incoming span ID was shared (from b3 headers)? if so, add span.shared=true

File: lib/zipkin-tracer/trace.rb [hunk 3]
```diff
@@ -199,12 +162,29 @@ def to_i; @i128; end
 
   # A span may contain many annotations
   class Span
-    attr_accessor :name, :annotations, :binary_annotations, :debug
+    module Tag
+      METHOD = "http.method".freeze
+      PATH = "http.path".freeze
+      STATUS = "http.status".freeze
+      LOCAL_COMPONENT = "lc".freeze
```
Comment (adriancole) [inline lib/zipkin-tracer/trace.rb]: ls is no longer needed unless there's a value for it. sometimes we had "lc"->"" just to add an endpoint to a span.

File: lib/zipkin-tracer/trace.rb [hunk 4]
```diff
@@ -199,12 +162,29 @@ def to_i; @i128; end
 
   # A span may contain many annotations
   class Span
-    attr_accessor :name, :annotations, :binary_annotations, :debug
+    module Tag
+      METHOD = "http.method".freeze
+      PATH = "http.path".freeze
+      STATUS = "http.status".freeze
```
Comment (adriancole) [inline lib/zipkin-tracer/trace.rb]: this is wrong, it should be http.status_code though possibly was always incorrect here

File: lib/zipkin-tracer/hostname_resolver.rb (hunk 1 above)
Comment (ykitamura-mdsol) [reply lib/zipkin-tracer/hostname_resolver.rb]: I think this comment is explaining:
https://github.com/openzipkin/zipkin-ruby/blob/f0435f983ca63644db0fdf911f5bb52b2b9df87b/lib/zipkin-tracer/hostname_resolver.rb#L5-L6

File: lib/zipkin-tracer/rack/zipkin-tracer.rb (hunk 2 above)
Comment (ykitamura-mdsol) [reply lib/zipkin-tracer/rack/zipkin-tracer.rb]: updated in https://github.com/openzipkin/zipkin-ruby/pull/141/commits/1139cc55956a570a48dca0230f25ac64ef2b7b2a

File: lib/zipkin-tracer/trace.rb (hunk 3 above)
Comment (ykitamura-mdsol) [reply lib/zipkin-tracer/trace.rb]: It seems we are still using it in our library. Added TODO in https://github.com/openzipkin/zipkin-ruby/pull/141/commits/2cd021ef2861135539effbcd8d9088c48d308a81

File: lib/zipkin-tracer/trace.rb (hunk 4 above)
Comment (ykitamura-mdsol) [reply lib/zipkin-tracer/trace.rb]: updated https://github.com/openzipkin/zipkin-ruby/pull/141/commits/d4b62540cef69d5c012ee05a2355e090339c1f56

**Comment by ykitamura-mdsol** [comment]
It seems we always have `duration` in spans:
https://github.com/openzipkin/zipkin-ruby/blob/76ea4940d019f03d32ec036f6f6a4e114bc79669/lib/zipkin-tracer/trace.rb#L213-L215

**Comment by jcarres-mdsol** [comment]
I am going to merge with the `lc` TODO. Moving to V2 is a huge step forward

**Comment by adriancole** [comment]
agreed. this is a huge help

**Pull request merged**